metadata-ingestion/src/datahub/ingestion/source/aws/dynamodb.py

```diff
@@ -0,0 +1,226 @@
+        super().__init__(ctx)
+        self.config = config
+        self.report = DynamoDBSourceReport()
+
+
+    @classmethod
+    def create(cls, source_config, context):
+        config = DynamoDBSourceConfig.parse_obj(source_config)
+        return cls(config, context)
+
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        if self.config.ingest_tables:
+            all_tables = self.config.ingest_tables
+        else:
+            all_tables = get_all_tables(self.config.dynamodb_client)
+        for table_name in all_tables:
+            logger.info(f"Processing table {table_name}")
+            if not self.config.table_pattern.allowed(table_name):
+                logger.info(f"Table {table_name} is not allowed by table_allow pattern")
+                self.report.report_table_filtered(table_name)
+                continue
+
+            wu = self.make_workunit(table_name)
+            if wu is None:
+                logger.info('Skip none work unit of table {table_name}')
+                continue
+            yield wu
+
+
+    def make_workunit(self, table_name: str):
+        table = describe_table(self.config.dynamodb_client, table_name)
+        dataset_urn = make_dataset_urn(DEFAULT_PLATFORM, table_name, self.config.env)
+        keys = [x["AttributeName"] for x in table.get("KeySchema", [])]
+
+        attribute_definitions = self.populate_attribute_definitions(table)
+        if attribute_definitions is None:
+            return None
+        schema_fields = self.get_schema_fields(attribute_definitions)
+        schema_metadata = SchemaMetadata(
+            schemaName=f"{table_name}-schema",
+            platform=make_data_platform_urn(DEFAULT_PLATFORM),
+            version=0,
+            hash="",
+            platformSchema=KeyValueSchema(keySchema="", valueSchema=""),
+            fields=schema_fields,
+            primaryKeys=keys
+        )
+
+        mcp = MetadataChangeProposalWrapper(
+            entityType="dataset",
+            entityUrn=dataset_urn,
+            changeType=ChangeTypeClass.UPSERT,
+            aspectName="schemaMetadata",
+            aspect=schema_metadata
+        )
+        wu = MetadataWorkUnit(
+            id=dataset_urn,
+            mcp=mcp
+        )
+        return wu
+
+
+    def populate_attribute_definitions(self, table) -> Dict[str, str]:
+        raw_attribute_definitions = table["AttributeDefinitions"]
+        attribute_definitions = {
+            x["AttributeName"]: x["AttributeType"] for x in raw_attribute_definitions
+        }
+        if table.get("StreamSpecification", {}).get("StreamEnabled", False):
+            stream_arn = table["LatestStreamArn"]
+            attribute_definitions.update(
+                get_attribute_definitions(self.config.dynamodbstreams_client, stream_arn)
+            )
+        else:
+            table_name = table["TableName"]
+            logger.warn(f"Table {table_name} streaming is NOT enabled.")
```

Comment:
I think error level logging makes sense here as we do want to fix the situation.